src/core/sequence.ml

```diff
@@ -0,0 +1,41 @@
+
+type +'a node =
+   | Nil
+   | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
+
+let rec map f i () = match i () with 
```

Comment:
Sorry, I didn't mean it's bad to use `match .. with`, just that the formatting is unfortunate because it looks very similar to this:
```ml
let rec map f i () = function
| Nil -> Nil
| Cons (x, r) -> Cons (f x, map f r)
```
I think it's bad practice more generally to place code of any significance at what is effectively a different indent level, but `match ... with` is especially unfortunate due to this similarity. So all I'm really asking is that you format it like this instead:
```ml
let rec map f i () =
  match i () with
  | Nil -> Nil
  | Cons (x, r) -> Cons (f x, map f r)
```